### ISSUE TITLE
Failed to mount volume when CSI Migration on k8s v1.17.1

### ISSUE THREAD

**Issue by celiawa** [kubernetes-sigs/azuredisk-csi-driver#309]
I'm using azuredisk csi 0.6.0. When do the CSI Migration test.
Found that the volume can be created and attached using the in-tree storageclass. But the pod failed to start because the volume can't be mount with below error.
```
  Type     Reason                  Age                  From                                     Message
  ----     ------                  ----                 ----                                     -------
  Normal   Scheduled               <unknown>            default-scheduler                        Successfully assigned default/cinfo-azuredisk-0 to azure-onekey-azure-onekey-w-01
  Normal   SuccessfulAttachVolume  6m43s                attachdetach-controller                  AttachVolume.Attach succeeded for volume "pvc-7cea0d8e-42e0-482f-b8ef-55d0b0d6ebd9"
  Warning  FailedMount             5m                   kubelet, azure-onekey-azure-onekey-w-01  Unable to attach or mount volumes: unmounted volumes=[azurediskpvc], unattached volumes=[default-token-glczb azurediskpvc]: timed out waiting for the condition
  Warning  FailedMount             25s (x2 over 2m42s)  kubelet, azure-onekey-azure-onekey-w-01  Unable to attach or mount volumes: unmounted volumes=[azurediskpvc], unattached volumes=[azurediskpvc default-token-glczb]: timed out waiting for the condition
  Warning  FailedMount             8s (x11 over 6m20s)  kubelet, azure-onekey-azure-onekey-w-01  MountVolume.WaitForAttach failed for volume "pvc-7cea0d8e-42e0-482f-b8ef-55d0b0d6ebd9" : volume /subscriptions/66e8f030-dbe5-4677-bd29-06b963faa0ac/resourceGroups/compaas-01-dev/providers/Microsoft.Compute/disks/pvc-7cea0d8e-42e0-482f-b8ef-55d0b0d6ebd9 has GET error for volume attachment csi-8f6f465b29e089269caecd4fb55b8015bbd9f2e0c9e4717b87a0b496e7dd5fec: volumeattachments.storage.k8s.io "csi-8f6f465b29e089269caecd4fb55b8015bbd9f2e0c9e4717b87a0b496e7dd5fec" is forbidden: User "system:node:azure-onekey-azure-onekey-w-01" cannot get resource "volumeattachments" in API group "storage.k8s.io" at the cluster scope: no relationship found between node "azure-onekey-azure-onekey-w-01" and this object
```

**Environment**:
- CSI Driver version: 0.6.0
- Kubernetes version (use `kubectl version`):   v1.17.1 
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

**Comment by andyzhangx** [comment]
not sure about the exact problem, looks like your deployment lacks of following rbac config?
https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/c0495bf9e17e10806b9f2f47b3bd968295a4b3d5/deploy/rbac-csi-azuredisk-controller.yaml#L72-L74
How did you set up such a cluster? by aks-engine? and also have you enabled `--feature-gates=CSIMigration=true,CSIMigrationAzureDisk=true` in kube-controller-manager?

**Comment by andyzhangx** [comment]
just let me know how can I repro this issue on my env. thanks.

**Comment by celiawa** [comment]
Checked that I have following rbac config.

  - apiGroups: ["storage.k8s.io"]
    resources: ["volumeattachments"]
    verbs: ["get", "list", "watch", "update"]

And I enabled the - --feature-gates=RotateKubeletServerCertificate=true,CSIMigration=true,CSIMigrationAzureDisk=true
in kube-controller-manager.yml and also in kubelet-config.yml.

The intree sc is as below.
[root@azure-onekey-azure-onekey-cs-01 templates]# kubectl describe sc azure-disk-standard0
Name:            azure-disk-standard0
IsDefaultClass:  Yes
Annotations:     kubectl.kubernetes.io/last-applied-configuration={"apiVersion":"storage.k8s.io/v1","kind":"StorageClass","metadata":{"annotations":{"storageclass.kubernetes.io/is-default-class":"true"},"name":"azure-disk-standard0"},"parameters":{"kind":"managed","storageaccounttype":"Standard_LRS"},"provisioner":"kubernetes.io/azure-disk"}
,storageclass.kubernetes.io/is-default-class=true
Provisioner:           kubernetes.io/azure-disk
Parameters:            kind=managed,storageaccounttype=Standard_LRS
AllowVolumeExpansion:  <unset>
MountOptions:          <none>
ReclaimPolicy:         Delete
VolumeBindingMode:     Immediate
Events:                <none>


The steps:
1. Enable CSI Migration
2.  Deploy a statefulset and  the PVC with the intree sc. 

[root@azure-onekey-azure-onekey-cs-01 templates]# kubectl get pvc
NAME                              STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS           AGE
azurediskpvc-cinfo-azuredisk-0    Bound    pvc-7cea0d8e-42e0-482f-b8ef-55d0b0d6ebd9   1Gi        RWO            azure-disk-standard0   115m


[root@azure-onekey-azure-onekey-cs-01 templates]# kubectl get pv |grep azure-disk-standard0
pvc-7cea0d8e-42e0-482f-b8ef-55d0b0d6ebd9     1Gi        RWO            Delete           Bound    default/azurediskpvc-cinfo-azuredisk-0    azure-disk-standard0            118m

[root@azure-onekey-azure-onekey-cs-01 templates]#  kubectl  describe pv pvc-7cea0d8e-42e0-482f-b8ef-55d0b0d6ebd9
Name:              pvc-7cea0d8e-42e0-482f-b8ef-55d0b0d6ebd9
Labels:            <none>
Annotations:       pv.kubernetes.io/provisioned-by: disk.csi.azure.com
Finalizers:        [kubernetes.io/pv-protection]
StorageClass:      azure-disk-standard0
Status:            Bound
Claim:             default/azurediskpvc-cinfo-azuredisk-0
Reclaim Policy:    Delete
Access Modes:      RWO
VolumeMode:        Filesystem
Capacity:          1Gi
Node Affinity:
  Required Terms:
    Term 0:        topology.disk.csi.azure.com/zone in []
Message:
Source:
    Type:         AzureDisk (an Azure Data Disk mount on the host and bind mount to the pod)
    DiskName:     pvc-7cea0d8e-42e0-482f-b8ef-55d0b0d6ebd9
    DiskURI:      /subscriptions/66e8f030-dbe5-4677-bd29-06b963faa0ac/resourceGroups/compaas-01-dev/providers/Microsoft.Compute/disks/pvc-7cea0d8e-42e0-482f-b8ef-55d0b0d6ebd9
    Kind:         Managed
    FSType:       ext4
    CachingMode:  ReadWrite
    ReadOnly:     false
Events:           <none>
 

3. Check the po, the pod failed to run because can't mount.

**Comment by celiawa** [comment]
The volumeattachment relevant role is added.

[root@azure-onekey-azure-onekey-cs-01 templates]# kubectl describe clusterroles azuredisk-external-attacher-role
Name:         azuredisk-external-attacher-role
Labels:       chart=csi-azuredisk
              chartVersion=1.0.3
              heritage=Tiller
              release=csi-azuredisk
              revision=1
Annotations:  <none>
PolicyRule:
  Resources                         Non-Resource URLs  Resource Names  Verbs
  ---------                         -----------------  --------------  -----
  leases.coordination.k8s.io        []                 []              [get list watch create update patch]
  persistentvolumes                 []                 []              [get list watch update]
  volumeattachments.storage.k8s.io  []                 []              [get list watch update]
  nodes                             []                 []              [get list watch]
  csinodeinfos.csi.storage.k8s.io   []                 []              [get list watch]

**Comment by andyzhangx** [comment]
hi @celiawa there is a migration issue on kubelet fixed in k8s `v1.18.0-beta.0`, `v1.17.4`.
The migration only works on disk volume provision stage, so in `v1.17.1`, pls remove `CSIMigrationAzureDisk=true` in kubelet config on agent.

Here are more details, we are planning to add migration e2e tests from v1.18.0:
https://github.com/kubernetes-sigs/azuredisk-csi-driver/pull/298#issuecomment-593409131

Let me know if you have any questions, thanks.

**Comment by celiawa** [comment]
Thanks Andy. We'll disable Migration this release.

**Comment by andyzhangx** [comment]
fixed in v1.17.4+
